modules/friendlytag.js

```diff
@@ -1222,69 +1544,159 @@ Twinkle.tag.callbacks = {
 				pageText = pageText.replace(miRegex, "$1" + tagText + "}}\n");
 				tagText = "";
 
-			} else if( params.group && groupableTags.length >= 2 ) {
+				addUngroupedTags();
+
+			} else if( params.group && !miTest && (groupableExistingTags.length + groupableTags.length) >= 2 ) {
 				Morebits.status.info( 'Info', 'Grouping supported tags inside {{multiple issues}}' );
 
-				groupableTags.sort();
-				tagText += '{{multiple issues|\n';
+				tagText += '{{Multiple issues|\n';
 
-				totalTags = groupableTags.length;
-				$.each(groupableTags, addTag);
+				/**
+				 * Adds newly added tags to MI
+				 */
+				var addNewTagsToMI = function() {
+					totalTags = groupableTags.length;
+					$.each(groupableTags, addTag);
+					if (groupableTags.length) {
+						summaryText += ' tags (within {{[[Template:multiple issues|multiple issues]]}})';
+					} else {
+						summaryText += ' {{[[Template:multiple issues|multiple issues]]}}'
+					}
+					if( tags.length > 0 ) {
+						summaryText += ', and';
+					}
+					tagText += '}}\n';
 
-				summaryText += ' tags (within {{[[Template:multiple issues|multiple issues]]}})';
-				if( tags.length > 0 ) {
-					summaryText += ', and';
+					addUngroupedTags();
+				};
+
+
+				var getRedirectsFor = [];
+
+				/**
+				 * Repositions the tag on the page into {{multiple issues}}, if found
+				 * with its proper name, else moves it to `getRedirectsFor` array to be
+				 * handled later
+				 * @param {string} tag
+				 */
+				var repositionTagIntoMI = function repositionTagIntoMI(tag) {
+					var tag_re = new RegExp('(\\{\\{' + Morebits.pageNameRegex(tag) + '\\s*(\\|[^}]+)?\\}\\}\\n?)');
+					if (tag_re.test(pageText)) {
+						tagText += tag_re.exec(pageText)[1];
+						pageText = pageText.replace(tag_re, '');
+					} else {
+						getRedirectsFor.push('Template:' + tag);
+					}
+				};
+
+				groupableExistingTags.forEach(repositionTagIntoMI);
+
+				if(! getRedirectsFor.length) {
+					addNewTagsToMI();
+				} else {
+					var api = new Morebits.wiki.api("Getting template redirects", {
```

Comment:
I am not sure if it would be a wise idea to try to pull out the common code into a function, as it would result in a lot of head-scratching to figure out the scope issues. Also, its repeated just once and there are some differences - the morebits.status.warn text, the MI one adds to tagText before removing, and the MI one calls addNewTagsToMI() at the end rather than postRemoval().

Reply:
That's a good enough explanation for me!